File: react-frontend/src/models/event.ts
import AccessTypes from "@/enums/accessTypes";

export interface Event {
    id: string | undefined;
    name: string;
    description: string;
    location: string;
    startDateTime: string;
    endDateTime: string;
    eventType: string;
    maxParticipants: number;
    accessType: AccessTypes;
    organizerId: string;
}

export interface EventType {
    id: string;
    name: string;
}

File: eventmanagement/src/main/java/webdev2/eventmanagement/model/Event.java
package webdev2.eventmanagement.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import webdev2.eventmanagement.model.enums.AccessType;

import java.time.LocalDateTime;

@Data
@Document(collection = "events")
public class Event {
    @Id
    private String id;
    private String name;
    private String description;
    private String location;
    private LocalDateTime startDateTime;
    private LocalDateTime endDateTime;
    private String eventType;
    private int maxParticipants;
    private AccessType accessType;
    private String organizerId;
}
